### PR TITLE
Observability: Update OpenTelemetry doc

### LINKED ISSUE #886: Observability: Update OpenTelemetry doc
**Is your feature request related to a problem? Please describe.**

In some [sections](https://github.com/microsoft/code-with-engineering-playbook/blob/main/docs/observability/tools/OpenTelemetry.md), the term `OpenTelemetry Agent` seems to refer directly to the OpenTelemetry java agent jar.
In case of Java, the java agent offers the automatic instrumentation but for other languages it’s not necessarily the “agent” that provides that.
Besides that, in the official OpenTelemetry documentation, the `OpenTelemetry Agent`  is defined as the [Collector deployment type](https://opentelemetry.io/docs/collector/deployment/)
and is not related to the automatic instrumentation that is available for some languages.
The nomenclature inconsistence between OpenTelemetry docs and information in the playbook can be confusing for readers.

The section [integration-options-with-azure-monitor](https://github.com/microsoft/code-with-engineering-playbook/blob/main/docs/observability/tools/OpenTelemetry.md#integration-options-with-azure-monitor) describes only Java use case as it compares two Java agents jar files available. But it's not mentioned explicitly that the section is relevant only for Java.
 
**Describe the solution you'd like**

- Update the OpenTelemetry doc to be more generic and language agnostic.
- Provide links to the OpenTelemetry documentation for more information/details on the OpenTelemetry concepts.
- Make the usage of `OpenTelemetry Agent` term consistent with the OpenTelemetry documentation

### PR DESCRIPTION
**Pull request by Izzzu**
# Pull Request Template

## What are you trying to address

This PR aims to make the OpenTelemetry doc language agnostic and more consistent with OpenTelemetry documentation for the naming and concept definitions. It also adds more links to the original OpenTelemetry documentation pages to make sure the most up to date information are referenced in the playbook.

Closes #886 

## Description of new changes

- Update the OpenTelemetry doc to be more generic and language agnostic.
- Provide links to the OpenTelemetry documentation for more information/details on the OpenTelemetry concepts.
- Make the usage of `OpenTelemetry Agent` term consistent with the OpenTelemetry documentation

## For all pull requests

- [x] Link to the issue you are solving (so it gets closed)
- [x] Label the pull request with the appropriate area(s)
- [x] Assign potential reviewers (you may also want to contact them on Teams to ensure timely reviews)
- [x] Assign the project - Engineering Playbook Backlog
- [x] Assign the pull request to yourself

### CODE REVIEW

**Review by TessFerrandez — APPROVED** [review]

**Review by shiranr — APPROVED** [review]

File: docs/observability/tools/OpenTelemetry.md [hunk 1]
```diff
@@ -73,27 +74,28 @@ From the website:
...
+Therefore, the main difference between running the OpenTelemetry Java Agent vs. the Application Insights Java Agent is demonstrated in the amount of traces getting logged in Azure Monitor. When running with Application Insights Java agent there's more telemetry getting pushed to Azure Monitor. On the other hand, when running the solution using the Application Insights agent mode, it is essential to highlight that nothing gets logged on Jaeger (or any other OpenTelemetry exporter). All traces will be pushed exclusively to Azure Monitor. However, both manual instrumentation done via the OpenTelemetry SDK and all automatic traces, dependencies, performance counters, and metrics being instrumented by the Application Insights agent are sent to Azure Monitor. Although there is a rich amount of additional data automatically instrumented by the Application Insights agent, it can be deduced that it is not necessarily OpenTelemetry compliant. Only the traces logged by the manual instrumentation using the OpenTelemetry SDK are.
 
-### OtelTelemetry vs Application Insights agents compared
+#### OtelTelemetry vs Application Insights agents compared
```
Comment (NarmathaBala) [inline docs/observability/tools/OpenTelemetry.md:98]:
```suggestion
#### OpenTelemetry vs Application Insights agents compared
```

File: docs/observability/tools/OpenTelemetry.md [hunk 2]
```diff
@@ -73,27 +74,28 @@ From the website:
...
 
 | Highlight                                                                | OtelTelemetry Agent | App Insights Agent |
```
Comment (NarmathaBala) [inline docs/observability/tools/OpenTelemetry.md:100]:
```suggestion
| Highlight                                                                | OpenTelemetry Agent | App Insights Agent |
```

**Review by NarmathaBala — APPROVED** [review]
Thanks @Izzzu for adding this. I have some minor suggestions

**Comment by Izzzu** [comment]
> Thanks @Izzzu for adding this. I have some minor suggestions

Thanks! I applied your suggestions

**Comment by shiranr** [comment]
@Izzzu can I merge this?

**Comment by Izzzu** [comment]
> Thank you for contributing @Izzzu, can I merge this?

Yes, please :)

**Pull request merged**